src/Analysis/Ast/Impl/Modules/Resolution/MainModuleResolution.cs

```diff
@@ -55,6 +55,18 @@ public MainModuleResolution(string root, IServiceContainer services, ImmutableAr
 
         public IBuiltinsPythonModule BuiltinsModule { get; private set; }
 
+        public IEnumerable<IPythonModule> GetImportedModules(CancellationToken cancellationToken) {
+            foreach (var moduleRef in Modules.Values) {
+                if (moduleRef.Value != null) {
```

Comment:
`cancellationToken.Throw` I guess although I don't think this is a long operation.